kiele

```diff
@@ -141,6 +150,34 @@ run_upload_report() {
     rm report.zip
 }
 
+run_compile() {
+    run_pyiele compile "$@"
+}
+
+run_testrunner() {
+    local run_file
+    check_listener
+    if [[ $# -gt 0 ]]; then
+      if [[ -f "$1" ]] || [[ -d "$1" ]]; then
+        run_file="$1"; shift
+        if [[ "$run_file" == *.py ]]; then
+            python3 "$run_file"
+        else
+            run_pyiele test --file "$run_file" "$@"
+        fi
+      else
+        run_pyiele test "$@"
+      fi
+    else
+      run_pyiele test "$@"
+    fi
+}
```

Comment:
You can still collapse those if blocks with this, otherwise it looks good to me.
```suggestion
run_testrunner() {
    local run_file cmd
    check_listener
    cmd=(run_pyiele test)
    if [[ $# -gt 0 ]] && ([[ -f "$1" ]] || [[ -d "$1" ]]); then
      run_file="$1"; shift
      if [[ "$run_file" == *.py ]]; then
        cmd=(python3 "$run_file")
      else
        cmd+=(--file "$run_file")
      fi
    fi
    "${cmd[@]}" "$@"
}
```